### PR TITLE
Fix spine can not show normally after disable and then active node.

### PR DESCRIPTION
**Pull request by bofeng-song**
Re: #
https://github.com/cocos/cocos-engine/issues/18008

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by dumganhar — APPROVED** [review]

**Comment by minggo** [comment]
What's the reason?

**Comment by bofeng-song** [comment]
> What's the reason?

After active unative is called, the buffer of the ts node is still used, but the native buffer is no longer updated, causing a rendering exception.

File: native/cocos/editor-support/MiddlewareManager.h [hunk 1]
```diff
@@ -108,16 +108,12 @@ class MiddlewareManager {
...
-    bool isUpdating = false;
-
 private:
-    void clearRemoveList();
+    void updateCache();
 
     ccstd::vector<IMiddleware *> _updateList;
-    ccstd::vector<IMiddleware *> _removeList;
-    std::map<int, MeshBuffer *> _mbMap;
+    ccstd::unordered_map<IMiddleware *, bool> _updateMap;
```
Comment (minggo) [inline native/cocos/editor-support/MiddlewareManager.h]: Please use a more suitable name, such as _removeMap

File: native/cocos/editor-support/MiddlewareManager.h [hunk 2]
```diff
@@ -108,16 +108,12 @@ class MiddlewareManager {
     MiddlewareManager();
     ~MiddlewareManager();
 
-    // If manager is traversing _updateMap, will set the flag untill traverse is finished.
-    bool isRendering = false;
-    bool isUpdating = false;
-
 private:
-    void clearRemoveList();
+    void updateCache();
```
Comment (minggo) [inline native/cocos/editor-support/MiddlewareManager.h]: Please use a more suitable name

File: native/cocos/editor-support/MiddlewareManager.cpp [hunk 3]
```diff
@@ -54,20 +54,20 @@ MeshBuffer *MiddlewareManager::getMeshBuffer(int format) {
...
-void MiddlewareManager::clearRemoveList() {
-    for (auto *editor : _removeList) {
-        auto it = std::find(_updateList.begin(), _updateList.end(), editor);
-        if (it != _updateList.end()) {
+void MiddlewareManager::updateCache() {
+    for (auto &iter: _updateMap) {
+        auto it = std::find(_updateList.begin(), _updateList.end(), iter.first);
+        if (iter.second && it == _updateList.end()) {
+            _updateList.push_back(iter.first);
+        } else if (!iter.second && it != _updateList.end()) {
```
Comment (minggo) [inline native/cocos/editor-support/MiddlewareManager.cpp]: - can polish the logic to
```c++
if (it != _updateList.end) {
    const toRemove = iter.second;
    if (toRemove) {
        ....
    } else {
        ....
    }
}
```

**Comment by minggo** [comment]
@cocos-robot run test cases

**Review by minggo — APPROVED** [review]

**Pull request merged**